plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -68,27 +68,43 @@ func isVolumeConflict(volume api.Volume, pod *api.Pod) bool {
 	if volume.AWSElasticBlockStore != nil {
 		volumeID := volume.AWSElasticBlockStore.VolumeID
 
-		manifest := &(pod.Spec)
-		for ix := range manifest.Volumes {
-			if manifest.Volumes[ix].AWSElasticBlockStore != nil &&
-				manifest.Volumes[ix].AWSElasticBlockStore.VolumeID == volumeID {
+		existingPods := &(pod.Spec)
+		for ix := range existingPods.Volumes {
+			if existingPods.Volumes[ix].AWSElasticBlockStore != nil &&
+				existingPods.Volumes[ix].AWSElasticBlockStore.VolumeID == volumeID {
 				return true
 			}
 		}
 	}
+	if volume.RBD != nil {
+		mon := volume.RBD.CephMonitors
+		pool := volume.RBD.RBDPool
+		image := volume.RBD.RBDImage
+
+		existingPods := &(pod.Spec)
+		for ix := range existingPods.Volumes {
+			if existingPods.Volumes[ix].RBD != nil {
+				mon_m := existingPods.Volumes[ix].RBD.CephMonitors
+				pool_m := existingPods.Volumes[ix].RBD.RBDPool
+				image_m := existingPods.Volumes[ix].RBD.RBDImage
+				if haveSame(mon, mon_m) && pool_m == pool && image_m == image {
+					return true
+				}
+			}
+		}
+	}
 	return false
 }
 
 // NoDiskConflict evaluates if a pod can fit due to the volumes it requests, and those that
-// are already mounted. Some times of volumes are mounted onto node machines.  For now, these mounts
-// are exclusive so if there is already a volume mounted on that node, another pod can't schedule
-// there. This is GCE and Amazon EBS specific for now.
+// are already mounted. If there is already a volume mounted on that node, another pod that uses the same volume
+// can't be scheduled there. This is GCE, Amazon EBS, and Ceph RBD specific for now.
```

Comment:
Add to comment that it currently checks GCE PD, AWS EBS, and Ceph RBD. Also mention that the policies are slightly different for each (GCE PD allows multiple mounts as long as they're all read-only, AWS EBS forbids any two pods mounting the same volume ID, Ceph RBD forbids if matching monitor, pool, and image.